### PR TITLE
HPCC-13863 Fix incorrect child filter, moving filter over project

### PR DESCRIPTION
**Pull request by ghalliday**
If a filter is moved over a project in a child query, and the transform for
the project also contains a reference to a value calculated from the input
dataset, it was possible for the filter that was being moved to be
incorrectly applied to the dataset inside the transform.

Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman here is the fix.
@jamienoss please review - this is a fairly critical bug.

File: testing/regress/ecl/issue13863.ecl [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems.
```
Comment (jamienoss) [inline testing/regress/ecl/issue13863.ecl]: 2015

File: testing/regress/ecl/issue13863.ecl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+r := { unsigned id1, unsigned id2 };
+ro := { unsigned id, DATASET(R) child };
```
Comment (jamienoss) [inline testing/regress/ecl/issue13863.ecl]: trivial: R -> r

**Comment by ghalliday** [comment]
@jamienoss have you finished reviewing?

**Comment by jamienoss** [comment]
@ghalliday No, not yet. Is it possible for you to write a single test into this regression example that collapses the whole thing to a single simple result, e.g. boolean?

**Comment by ghalliday** [comment]
@jamienoss I could.  The test shows the problem - run before and after building your system.

**Comment by jamienoss** [comment]
@richardkchapman this looks good to merge

**Pull request merged**